data/blog/gn.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "Using GN to Build Electron"
+author: nornagon
+date: '2018-09-05'
+---
+
+Electron now uses GN to build itself. Here's a discussion of why.
+
+---
+
+# GYP and GN
+
+When Electron was first released in 2013, Chromium's build configuration was written with [GYP], short for "Generate Your Projects".
+
+In 2014, the Chromium project introduced a new build configuration tool called [GN] (short for "Generate [Ninja]") Chromium's build files were migrated to GN and GYP was removed from the source code.
+
+Electron has historically kept a separation between the main [Electron code] and [libchromiumcontent], the part of Electron that wraps Chromium's 'content' submodule. Electron has carried on using GYP, while libchromiumcontent -- as a subset of Chromium -- switched to GN when Chromium did.
+
+Like gears that don't quite mesh, there was friction between using the two build systems. Maintaining compatibility was error-prone, from compiler flags and `#defines` that needed to be meticulously kept in sync between Chromium, Node, V8, and Electron.
+
+To address this, the Electron team has been working on moving everything to GN. Today, the [commit](https://github.com/electron/electron/pull/14097) to remove the last of the GYP code from Electron was landed in master.
+
+# What this means for you
+
+If you're contributing to Electron itself, the process of checking out and building Electron is very different. See [GN build instructions] for details.
+
+If you're developing an app with Electron, there are a few minor changes you might notice in the new Electron 4.0.0-nightly; but more than likely, Electron's change in build system will be totally transparent to you.
+
+# What this means for Electron
+
+GN is [faster](https://chromium.googlesource.com/chromium/src/tools/gn/+/48062805e19b4697c5fbd926dc649c78b6aaa138/README.md) than GYP and its files are more readable and maintainable. Moreover, we hope that using a single build configuration system will reduce the work required to upgrade Electron to new versions of Chromium.
```

Comment:
The link should probably point to the new GN repo, e.g. https://gn.googlesource.com/gn/+/master/README.md

Reply:
Yes, that is a good link! I've put it in where GN is introduced at the top of the post.

Since the context here is about comparing GN to GYP, the link to the Chromium thread discussing GN as being faster than GYP makes sense here.